src/Path.hs

```diff
@@ -110,6 +110,7 @@ data PathParseException
   | InvalidAbsFile FilePath
   | InvalidRelFile FilePath
   | Couldn'tStripPrefixDir FilePath FilePath
+  | NoParent
```

Comment:
How about the name of the exception type? `PathParseException` is a misnomer with the strip prefix and now noparent stuff added to it.